### PR TITLE
refactor!: Replace SecretStore service config with default values and overrides

### PR DESCRIPTION
**Pull request by lenny-intel**
BREAKING CHANGE: SecretStore config no longer in service configuration file. Changes must be done via use of environment variable overrides of default values

Signed-off-by: Leonard Goodell <leonard.goodell@intel.com>

If your build fails due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/go-mod-bootstrap/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [x] I have added unit tests for the new feature or bug fix (if not, why?)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  TBD

## Testing Instructions
Build core data with this branch
Run secure EdgeX stack, store core data container and remove core data config from Consul
Run core data locally with DEBUG logging and SecretStore enabled.
```
sudo EDGEX_SECURITY_SECRET_STORE=true WRITABLE_LOGLEVEL=DEBUG ./core-data -cp -r
```
Verify core data bootstraps successfully and is receiving event from MessageBus
Verify core data config loaded from file and pushed to Consul 
Verify core data config in Consul no longer contains SecretStore section
Restart core data
Verify core data config loaded from Consul, never from file. 

More extensive testing will occur when edgex-go consumes these changes.

### CODE REVIEW

File: bootstrap/bootstrap.go [hunk 1]
```diff
@@ -17,17 +17,19 @@ package bootstrap
...
 import (
 	"context"
+	"fmt"
 	"os"
 	"os/signal"
 	"sync"
 	"syscall"
 
-	"github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/config"
+	bootstrapConfig "github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/config"
```
Comment (bnevis-i) [inline bootstrap/bootstrap.go:26]: Don't see why need to do this -- there are no other "config" imports in this file.

**Review by bnevis-i — APPROVED** [review]
LGTM

File: bootstrap/secret/secret_test.go [hunk 2]
```diff
@@ -81,7 +79,7 @@ func TestNewSecretProvider(t *testing.T) {
 					case "/v1/auth/token/lookup-self":
 						w.WriteHeader(http.StatusOK)
 						_, _ = w.Write([]byte(testTokenResponse))
-					case "/redisdb":
+					case "/v1/secret/edgex/testServiceKey//redisdb":
```
Comment (jim-wang-intel) [inline bootstrap/secret/secret_test.go:82]: remove one slash:
```suggestion
					case "/v1/secret/edgex/testServiceKey/redisdb":
```

Comment (lenny-intel) [reply bootstrap/secret/secret_test.go:82]: Nope, this is how it currently gets generated by code in go-mod-secrets. I tried to use single slash... :-(

Comment (jim-wang-intel) [reply bootstrap/secret/secret_test.go:82]: ok

**Review by jim-wang-intel — APPROVED** [review]

File: bootstrap/bootstrap.go (hunk 1 above)
Comment (lenny-intel) [reply bootstrap/bootstrap.go]: Not needed. Removed.

**Pull request merged**